Sources/MapboxMaps/Style/Style+Localization.swift

```diff
@@ -0,0 +1,83 @@
+@_implementationOnly import MapboxCommon_Private
+
+/// Support for localization of labels
+extension Style {
+
+    /// This function creates an expression that will localize the `textField` property of a `SymbolLayer`
+    /// - Parameter locale: A `SupportedLanguage` based `Locale`
+    internal func localizeLabels(into locale: Locale) throws {
+
+        /// Get all symbol layers that are currently on the map
+        let symbolLayers = allLayerIdentifiers.filter { layer in
+            return layer.type == .symbol
+        }
+
+        /// Expression to be applied to the `SymbolLayer.textField`to localize the language
+        /// Sample Expression JSON: `["format",["coalesce",["get","name_en"],["get","name"]],{}]`
+        let replacement = "[\"get\",\"name_\(getLocaleValue(locale: locale))\"]"
+
+        let expressionCoalesce = try NSRegularExpression(pattern: "\\[\"get\",\\s*\"(name_.{2,7})\"\\]",
+                                                         options: .caseInsensitive)
+        let expressionAbbr = try NSRegularExpression(pattern: "\\[\"get\",\\s*\"abbr\"\\]",
+                                                     options: .caseInsensitive)
```

Comment:
It’s unnecessary to convert the entire expression to JSON, then to a string, then monkeypatch the string. That defeats the purpose of having a structured, type-safe representation of an expression.

Consider adapting the implementation in mapbox/mapbox-navigation-ios#2933 that recurses into the expression, surgically replacing occurrences of `name` or `name_*` with the coalescing expression. That approach is much more reliable and maintainable than trying to account for the variety of possible `text-field` values with finding-and-replacing in source code.